src/internal_macros.rs

```diff
@@ -282,6 +282,28 @@ macro_rules! serde_struct_impl {
                         formatter.write_str("a struct")
                     }
 
+                    fn visit_seq<V>(self, mut seq: V) -> Result<Self::Value, V::Error>
+                    where
+                        V: $crate::serde::de::SeqAccess<'de>,
+                    {
+                        use $crate::serde::de::Error;
+
+                        let length = 0;
+                        $(
+                            let $fe = seq.next_element()?.ok_or_else(|| {
+                                Error::invalid_length(length, &self)
+                            })?;
+                            #[allow(unused_variables)]
+                            let length = length + 1;
+                        )*
```

Comment:
Did you look at the expanded output? I'm not sure if the length is preserved the way you'd want it.
could you replace it with a mutable length that you add 1 to?

Reply:
This works fine the way it is. It's the way I usually write these sorts of macros. I can change it if you want though.

Reply:
It looks like it'd work, but it seems like a naive compiler would recreate the length variable at the top of the stack every time instead of just incrementing. Why can't you just drop the `let`? `length += 1;` should do even.

Reply:
Do we really need to worry about compiler bugs producing incorrect code?

I tried changing it anyway, since it's not worth arguing about, but then you can't add the `#[allow(unused_assignment)]` attribute to the line because rust complains that "attributes on expressions are experimental", and without that you get compiler warnings about the last incrementation of `length` being unused.